File: 30 days of code/__ Java/Day 19 - Interfaces.java
/**
 *              INTERFACES | JAVA
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 19
 */

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws FileNotFoundException {
        File file = new File("input.txt");
        Scanner scan = new Scanner(file);
        {
            int n = scan.nextInt();

            IAdvancedArithmetic calculator = new Calculator();
            System.out.println("I implemented: " + calculator.getClass().getInterfaces()[0].getName());

            int sum = calculator.divisorSum(n);
            System.out.println(sum);
        }
        scan.close();
    }
}

interface IAdvancedArithmetic {
    int divisorSum(int n);
}

class Calculator implements IAdvancedArithmetic {
    public int divisorSum(int n) {
        int sum = n;
        for (int i = 1; i <= n / 2; ++i) {
            sum += (n % i == 0) ? i : 0;
        }

        return sum;
    }
}

File: 30 days of code/__ C#/Day 19 - Interfaces.cs
/**
 *              INTERFACES | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 19
 */

using System;
using System.IO;

class Program
{
    static void Main(string[] args)
    {
        var reader = new InputReader(false);
        int n = reader.NextInt;
        reader.Close();

        IAdvancedArithmetic calculator = new Calculator();
        int sum = calculator.DivisorSum(n);

        Console.WriteLine("I implemented: " + calculator.GetType().GetInterfaces()[0].Name);
        Console.WriteLine(sum);
    }
}

interface IAdvancedArithmetic
{
    int DivisorSum(int n);
}

class Calculator : IAdvancedArithmetic
{
    public int DivisorSum(int n)
    {
        int sum = n;

        for (int i = 1; i <= n / 2; ++i)
        {
            sum += (n % i == 0) ? i : 0;
        }

        return sum;
    }
}

class InputReader
{
    private TextReader input;

    public InputReader(bool fromFile)
    {
        input = fromFile ? new StreamReader("input.txt") : Console.In;
    }

    public string NextLine => input.ReadLine();

    public int NextInt => int.Parse(NextLine);

    public void Close() => input.Close();
}
